packages/kit/src/exports/public.d.ts

```diff
@@ -42,14 +43,26 @@ export interface Adapter {
 		/**
 		 * Test support for `read` from `$app/server`
 		 * @param config The merged route config
+		 * @param route The route and its ID
+		 * @param entry Name of the entry point, in case this was called from an additional entry point (route and config are irrelevant in this case)
 		 */
-		read?: (details: { config: any; route: { id: string } }) => boolean;
+		read?: (details: { config: any; route: { id: string }; entry?: string }) => boolean;
 	};
 	/**
 	 * Creates an `Emulator`, which allows the adapter to influence the environment
 	 * during dev, build and prerendering
 	 */
-	emulate?: () => MaybePromise<Emulator>;
+	emulate?: (helpers: {
+		/** Allows to import an entry point defined within `additionalEntryPoints` by referencing its name */
+		importEntryPoint: (name: string) => Promise<any>;
+	}) => MaybePromise<Emulator>;
+	/**
+	 * An object with additional entry points for Vite to consider during compilation.
+	 * The key is the name of the entry point that will be later available at `${builder.getServerDirectory()}/adapter/<name>.js`,
+	 * the value is the relative path to the entry point file.
+	 * This is useful for adapters that want to generate separate bundles for e.g. middleware.
```

Comment:
```suggestion
	 * Additional entry points that will be bundled by Vite.
	 * The key is the name of the entry point. It can be used with `importEntryPoint(name)` inside `emulate`, and the resulting module is available inside `adapt` as `${builder.getServerDirectory()}/adapter/<name>.js`.
	 * The value is the path to the entry point, relative to the current working directory.
	 * This is useful for adapters that want to generate separate bundles for e.g. middleware.
```